### DISCUSSION TITLE
Firmware download for C720 / TC55?

### DISCUSSION THREAD

**Discussion by airdrummingfool** [JurajNyiri/HomeAssistant-Tapo-Control#707] [Discuss]
This question isn't really specific to this repo, but there are probably a good amount of people here that are familiar with firmware downgrading and whatnot so I figured it would be a good place to post. My apologies if it is inappropriate.

Does anyone know where I can download the firmware files for the Tapo C720 / TC55 floodlight camera? I've looked through the `all_keys.txt` file linked in [this discussion](https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/discussions/625#discussion-6941269),  but the last commit was over a year ago and the C720 is a new camera; there's nothing listed for it.

Somewhat relatedly, there is a new firmware update for the floodlight cam (I am hoping to get a download of the old version before I try this new one out, in case it is worse). From their [TC55 download page](https://www.tp-link.com/us/support/download/tc55/#Firmware-Release-Notes):

```
TC55(US)_V1.6_1.2.3 Build 240823
--
Published Date: 2024-11-05
Modifications and Bug Fixes:
1. Added support for 12-hour Time option and more timestamp formats.
2. Enhanced stability of the camera.
3. Improved performance of the camera.
4. Fixed some minor bugs.
```

**Comment by PetePeter** [comment]
allkeys.txt has one on line 9846 to 9850
You mean none of them work?

please please let me know if they work, if so, then i'll buy 4x for myself as it's been something ive been wanting to do but i need the 2way audio to work on these ones.

If you do manage to get it to work please please tell me whether floodlight control works independent of the camera via HA.

Thank you.

**Reply by airdrummingfool** [reply]
> allkeys.txt has one on line 9846 to 9850

Ah!
```
firmware/Tapo_C720v1_en_1.0.15_Build_230309_Rel.45493n_up_boot-signed_1680511254788.bin
firmware/Tapo_C720v1_en_1.0.15_Build_230309_Rel.45493n_up_boot-signed_1691632492511.bin
firmware/Tapo_C720v1_en_1.0.15_Build_230309_Rel.45493n_up_boot-signed_1691662050066.bin
firmware/Tapo_C720v1_en_1.0.15_Build_230309_Rel.45493n_up_boot-signed_1692170548072.bin
firmware/Tapo_C720v1_en_1.0.15_Build_230309_Rel.45493n_up_boot-signed_1692170680796.bin
```

Thanks for pointing those lines out, as I had not seen them. They do appear to be very old firmware versions (current is `1.2.3`, previous is `1.2.2`, this is `1.0.15`) and they might be specific to a previous hardware version of the C720 ("v1", which may or may not be different than my "v1.6"). Since they're so old I don't think there's much to be gained by trying them. Hopefully the list will be updated at some point with a link to `1.2.2` or `1.2.3`.

> If you do manage to get it to work please please tell me whether floodlight control works independent of the camera via HA.

I can turn the floodlight on from within HA, and if HA turns it on it can turn it off. However, I cannot turn off the floodlight from HA if it turns on due to motion detection. I don't really understand this limitation, and it's not clear if it's a problem in HA or the camera itself (or even my HA connection to the camera). Hopefully this is helpful!

**Reply by PetePeter** [reply]
thankyou
so, you turn off on the light from ha using what integration? this same one but when you onboard the camera without internet or with some other integration?

**Reply by airdrummingfool** [reply]
I'm using the integration in this repo (https://github.com/JurajNyiri/HomeAssistant-Tapo-Control). It is the only integration I use to interact with the camera.

**Reply by PetePeter** [reply]
ok so it works without downgrading if you onboard the camera with the internet blocked..right?

**Reply by airdrummingfool** [reply]
yes. It took me a few tries, and for the attempt that worked I actually had to hard reset it twice before connecting. But yes, I was able to set up this integration with my TC55 with firmware 1.2.2 with the camera's internet blocked. My understanding is that you have to keep the camera's internet blocked permanently, or the HA integration will eventually not be able to connect.

**Reply by PetePeter** [reply]
awesome!
Now i need to think whether i want to buy them right now during black friday sales at $184 AUD, or wait 7months when i'll actually need them for my new house.

**Reply by PetePeter** [reply]
yep my c720 works! bought two already. two more to go

**Reply by PetePeter** [reply]
I'm not updating firmware ever
